### PR TITLE
chore: add node version check to fail gracefully

### PR DESCRIPTION
**Pull request by raymondfeng**
Check required Node.js versions and print out meaningful error message if fails.

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

**Comment by bajtos** [comment]
I find it difficult to review this change because there is no easy way how to show possible changes made while moving most of the code from `bin/cli.js` to `bin/main.js`. This also looses git history of the changes made in this part of CLI code base, rendering `git blame` rather useless.

Have you considered creating a new file, e.g. `cli-with-check.js` or `cli-entry.js` that would contain the newly-added version-check, and keep most of the existing code inside `bin/cli.js` with no (or very little) changes?

**Comment by raymondfeng** [comment]
@bajtos I created `bin/cli-main.js` for `lb4` and moved `bin/cli.js` to `lib/cli.js`. Version histories are now preserved. PTAL.

File: packages/dist-util/index.js [hunk 1]
```diff
@@ -6,15 +6,28 @@
...
 const path = require('path');
+const util = require('util');
+const semver = require('semver');
+
+/**
+ * Make sure node version meets the requirement. This file intentionally
+ * only uses ES5 features so that it can be run with lower versions of Node
+ * to report the version requirement.
+ */
+function checkNodeVersion(range) {
```
Comment (bajtos) [inline packages/dist-util/index.js]: AFAICT, `range` is not used anywhere. Should we remove it and simplify the implementation a tiny bit?

Comment (raymondfeng) [reply packages/dist-util/index.js]: It's used here - https://github.com/strongloop/loopback-next/blob/ece15edca2fb3fd5c177967b737bdc9e54365647/packages/cli/bin/cli-main.js#L12-L13

**Pull request merged**